### ISSUE TITLE
[BONUS] Implement ML-ready data pipeline and endpoints

### ISSUE THREAD

**Issue by fabricio-entringer** [fabricio-entringer/6mlet-tech-challenge-01#32]
## Description
Implement ML-ready data pipeline endpoints and features to support machine learning workflows as specified in the Tech Challenge bonus challenges.

## Requirements
- Create endpoints specifically designed for ML feature engineering
- Implement data export formats suitable for ML training
- Add endpoints for receiving and storing ML predictions
- Design feature extraction and data preparation utilities
- Create training dataset generation capabilities
- Implement data versioning and ML experiment tracking support
- Add performance monitoring for ML-ready features

## Acceptance Criteria
- [ ] GET /api/v1/ml/features - Data formatted for ML features
- [ ] GET /api/v1/ml/training-data - Complete dataset for training
- [ ] POST /api/v1/ml/predictions - Endpoint to receive predictions
- [ ] Feature engineering utilities for book recommendations
- [ ] Data export in multiple formats (JSON, CSV, Parquet)
- [ ] Data versioning and snapshot capabilities
- [ ] ML model metadata storage and retrieval
- [ ] Performance monitoring for ML endpoints
- [ ] Documentation for ML integration workflows
- [ ] Example notebooks for data scientists

### GET /api/v1/ml/features
```json
{
  \"features\": [
    {
      \"book_id\": \"book_001\",
      \"price_normalized\": 0.45,
      \"rating_scaled\": 0.8,
      \"category_encoded\": [0, 1, 0, 0, 0],
      \"availability_binary\": 1,
      \"title_features\": {...},
      \"created_at\": \"2025-06-09T09:00:00Z\"
    }
  ],
  \"feature_schema\": {...},
  \"version\": \"v1.0\"
}
```

### GET /api/v1/ml/training-data
```json
{
  \"dataset\": [...],
  \"metadata\": {
    \"total_samples\": 1000,
    \"features_count\": 15,
    \"target_variable\": \"rating\",
    \"split_info\": {...}
  },
  \"data_quality\": {...}
}
```

### POST /api/v1/ml/predictions
```json
{
  \"model_id\": \"book_recommender_v1\",
  \"predictions\": [
    {
      \"book_id\": \"book_001\",
      \"predicted_rating\": 4.2,
      \"confidence\": 0.85,
      \"timestamp\": \"2025-06-09T09:00:00Z\"
    }
  ]
}
```

## ML Features
- Feature engineering pipeline for book attributes
- Text feature extraction from titles and descriptions
- Categorical encoding for categories
- Numerical feature scaling and normalization
- Time-based features for data freshness
- Statistical features per category
- Recommendation system support features

## Data Science Integration
- Jupyter notebook examples for common ML workflows
- Python SDK for easy API integration
- Data quality reports and monitoring
- Feature importance tracking
- Model performance monitoring hooks

## Dependencies
- Requires core API endpoints to be implemented
- Requires data processing pipeline
- May require additional ML libraries (scikit-learn, pandas)

## Priority
Medium - **BONUS CHALLENGE** (additional points)

## Technical Requirements
- Support for batch and real-time feature serving
- Efficient data serialization formats
- Caching for frequently requested feature sets
- Integration with popular ML frameworks
- Data governance and lineage tracking

**Comment by fabricio-entringer** [comment]
@adrianoribeiro Plese, check if this issue can be closed.

**Comment by adrianoribeiro** [comment]
issue? I cannot see any issue here
